var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -30,9 +30,9 @@ class Lammps(CMakePackage):
     """LAMMPS stands for Large-scale Atomic/Molecular Massively
     Parallel Simulator."""
     homepage = "http://lammps.sandia.gov/"
-    url      = "https://github.com/lammps/lammps/archive/stable_01Sep2017.tar.gz"
+    url      = "https://github.com/lammps/lammps/archive/patch_01Sep2017.tar.gz"
```

Comment:
(this is not an error but IMO may create some confusion)

As of now it looks like it would be confusing for a user to add a version prefixed with "stable". IMO one of the following two changes would make this easier:

* remove `Lammps.url` and `Lammps.url_for_version` and define `url` in the `version()` declaration like:

```
    version('20170901', '767e7f07289663f033474dfe974974e7',
        url='https://github.com/lammps/lammps/archive/patch_1Sep2017.tar.gz')
```

* Redefine the version to include stable/patch and rewrite `url_for_version` like: 

```
    version('patch_20170901', '767e7f07289663f033474dfe974974e7')

    def url_for_version(self, version):
        prefix, datestr = version[0], version[1:]
        vdate = dt.datetime.strptime(str(datestr), "%Y%m%d")
        return "https://github.com/lammps/lammps/archive/{0}_{1}.tar.gz".format(
            prefix, vdate.strftime("%d%b%Y").lstrip('0'))
```

The first is easier (and IMO sufficient). If you think neither is a good approach, IMO there should at least be a comment suggesting that the user can declare `version(..., url=...)` for versions prefixed with "stable"

Reply:
I should have said that, each _stable_ release is also tagged as a _patch_ release, so switch to patch just gives our users a more fine-grained version control.

Reply:
We could only the store the checksums of the _stable_  release though, but in my experience lammps have a very good idea, which version they need.

Reply:
> I should have said that, each stable release is also tagged as a patch release

is it? Maybe I misunderstood what you mean.

Looking at https://github.com/lammps/lammps/releases there is a mixture of both `stable_` and `patch_` prefixes with different dates. Probably `stable` are major_minor releases whereas `patch` are bugfix releases. 

So I guess we would need to provide both. In that case, I would say the first option suggested by @scheibelp 
```
version('20170901', '767e7f07289663f033474dfe974974e7',
        url='https://github.com/lammps/lammps/archive/patch_1Sep2017.tar.gz')
```
is the way to go.

Reply:
btw, this

> version('patch_20170901', '767e7f07289663f033474dfe974974e7')

should rather be 

```
version('20170901_patch', '767e7f07289663f033474dfe974974e7')
```

so that `20170901_patch < 20171212_stable`.

This is also more convenient for users as they would immediately see if the version is among stable or patches.

Reply:
I meant e.g. `stable_11Aug2017` and `patch_11Aug2017` both point to lammps/lammps@0b3f1b8a156db65b5f3b8c01a7a10deaae119f81, see https://github.com/lammps/lammps/releases

Reply:
> I meant e.g. stable_11Aug2017 and patch_11Aug2017 both point to

indeed, i somehow overlooked that they are duplicated. There exist `stable_11Aug2017` and `patch_11Aug2017` pointing to the same commit. I checked a few others and they are all seem to be consistent in this.

Reply:
> We could mark the patch release as unstable, but I find them very useful when use a more cutting-edge feature.

patched releases should certainly be available, it's just not obvious for the user which one is stable and which one is a patch. Whereas it would be obvious if you adopt 
```
20170901.patch
20170811.stable
```
convention and mix URL's with `stable_`  and `patch_` via a custom `url_for_version`.

Reply:
Yeah, Steve uses a script to release lammps and hence it should be consistent.

Is there now a ways to mark versions as unstable even though listing their checksums?

Reply:
> Is there now a ways to mark versions as unstable even though listing their checksums?

`version(..., preferred=True)`. But this does not let users know which older version is stable. So I would still prefer encoding this in a version name and additionally do `

Reply:
Can we push this until there is stable version with CMake? (Open an issue as a reminder for me)

Reply:
Instead of encoding it in the version could we make a stable variant, which can only be satisfied by stable `when`?